x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/errors_types/post_index_error.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { MessageBody } from '../../../../components/ml/api/throw_if_not_ok';
+
+export class PostSignalError extends Error {
+  message: string = '';
+  statusCode: number = -1;
+  error: string = '';
+
+  constructor(errObj: MessageBody) {
+    super(errObj.message);
+    this.message = errObj.message ?? '';
+    this.statusCode = errObj.statusCode ?? -1;
+    this.error = errObj.error ?? '';
```

Comment:
nit: As with `SignalIndexError` above, you can reference the super state so it's not duplicated (and include `name`)

``` ts
this.message = errObj.message ?? this.message;
this.statusCode = errObj.statusCode ?? this.statusCode;
this.error = errObj.error ?? this.error;
this.name: 'PostSignalError';
```